File: medplum-java/medplum-model/src/main/java/com/fhir/CoverageEligibilityRequest_Item.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableCoverageEligibilityRequest_Item.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface CoverageEligibilityRequest_Item {
  Optional<Reference> provider();

  Optional<Reference> facility();

  Optional<Quantity> quantity();

  Optional<List<CodeableConcept>> modifier();

  Optional<List<CoverageEligibilityRequest_Diagnosis>> diagnosis();

  Optional<Money> unitPrice();

  Optional<List<positiveInt>> supportingInfoSequence();

  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Reference>> detail();

  Optional<CodeableConcept> category();

  Optional<CodeableConcept> productOrService();

  Optional<String> id();

  static ImmutableCoverageEligibilityRequest_Item.Builder builder() {
    return ImmutableCoverageEligibilityRequest_Item.builder();
  }
}

File: medplum-typescript/src/model/fhir/CoverageEligibilityRequest_Item.ts
import { CodeableConcept } from "./CodeableConcept";
import { CoverageEligibilityRequest_Diagnosis } from "./CoverageEligibilityRequest_Diagnosis";
import { Extension } from "./Extension";
import { Money } from "./Money";
import { positiveInt } from "./positiveInt";
import { Quantity } from "./Quantity";
import { Reference } from "./Reference";

export interface CoverageEligibilityRequest_Item {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    supportingInfoSequence: positiveInt[] | null | undefined;
    category: CodeableConcept | null | undefined;
    productOrService: CodeableConcept | null | undefined;
    modifier: CodeableConcept[] | null | undefined;
    provider: Reference | null | undefined;
    quantity: Quantity | null | undefined;
    unitPrice: Money | null | undefined;
    facility: Reference | null | undefined;
    diagnosis: CoverageEligibilityRequest_Diagnosis[] | null | undefined;
    detail: Reference[] | null | undefined;
}
